### PR TITLE
Ensure that schema hashes don't collide in a couple of different cases:

### PR DESCRIPTION
**Pull request by shans**
* different inline schemas with the same name are used
* schema fields differ only in depth of nesting inside inline schemas
* also add a couple of extra tests to schema normalized string generation

This change should fix b/167648998 as the root cause of this looks
to have been two schemas aliased together due to different inline
schemas with the same name.

### CODE REVIEW

File: src/types/internal/schema.ts [hunk 1]
```diff
@@ -454,7 +454,10 @@ export class Schema {
     } else if (kind === 'schema-tuple') {
       return `(${types.map(t => t.type).join('|')})`;
     } else if (kind === 'schema-ordered-list') {
-      return `List<${schema.type}>`;
+      if (schema.kind === 'schema-primitive' || schema.ind === 'kotlin-primitive') {
```
Comment (galganif) [inline src/types/internal/schema.ts:457]: schema.ind -> schema.kind?

File: src/types/internal/schema.ts [hunk 2]
```diff
@@ -466,6 +469,6 @@ export class Schema {
     return this.names.slice().sort().join(' ') + '/' +
       Object.keys(this.fields).sort().map(field =>
         `${field}:${this.normalizeTypeForHash(this.fields[field])}`
-      ).join('');
+      ).join('') + '/';
```
Comment (galganif) [inline src/types/internal/schema.ts:472]: I wonder if we are also better off having a separator between fields eg
).join('-') + '/';

**Review by galganif — APPROVED** [review]

File: src/types/internal/schema.ts (hunk 2 above)
Comment (shans) [reply src/types/internal/schema.ts:472]: there is a separator, of sorts, for fields that don't wrap (the vertical pipe). Perhaps it's better to remove it from the per-field export and do it systematically?

I'm also considering just retiring this method entirely and modifying the toString() method to take an optional sort parameter. Seems crazy that we have a guaranteed-unique representation for parsing that we don't use.

File: src/types/internal/schema.ts (hunk 1 above)
Comment (shans) [reply src/types/internal/schema.ts:457]: yikes, nice catch.

File: src/types/internal/schema.ts (hunk 2 above)
Comment (shans) [reply src/types/internal/schema.ts:472]: I've left as-is for now, (updating the hashes in all our tests is painful and effectively there is already a separator). I've filed the idea of moving to toString() as an issue (b/168267507).

**Comment by shans** [comment]
(g3 success: cl/330673385)

**Pull request merged**